rust/main/agents/relayer/src/merkle_tree/processor.rs

```diff
@@ -87,13 +94,23 @@ impl MerkleTreeProcessor {
             trace!(leaf_index=?self.leaf_index, "No merkle tree insertion found in DB for leaf index, waiting for it to be indexed");
             None
         };
+
+        self.metrics
+            .merkle_tree_retrieve_insertion_total_elapsed_micros
+            .inc_by(begin.elapsed().as_micros() as u64);
+        self.metrics.merkle_tree_retrieve_insertions_count.inc();
```

Comment:
should these be moved inside the `if` block along with the other metric update?

Reply:
Fixed